src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/OnboardingSupervisor.kt

```diff
@@ -73,21 +84,61 @@ internal class OnboardingSupervisor(
 
     private fun retrieveAssetsFromRouter() {
         if (StatsigConfig.useSkip) {
-            retrieveSkipTransferChains()
+            MainScope().launch {
+                retrieveSkipTransferChains()
+            }
             retrieveSkipTransferTokens()
         } else {
             retrieveTransferAssets()
         }
         retrieveCctpChainIds()
     }
 
-    private fun retrieveSkipTransferChains() {
+
+
+    private suspend fun retrieveSkipTransferChains() = coroutineScope {
+        // kick off rpc fetch in parallel with chains fetch
+        val retrieveRpcEndpointsDeferred = async { retrieveChainRpcEndpoints() }
+
         val oldState = stateMachine.state
         val chainsUrl = helper.configs.skipV1Chains()
-        helper.get(chainsUrl, null, null) { _, response, httpCode, _ ->
-            if (helper.success(httpCode) && response != null) {
-                update(stateMachine.routerChains(response), oldState)
+        val chainsRequestDeferred = async {
+            suspendCancellableCoroutine<String?> { continuation ->
+                helper.get(chainsUrl, null, null) { _, response, httpCode, _ ->
+                    if (helper.success(httpCode) && response != null) {
+                        continuation.resume(response)
+                    }
+                }
+            }
+        }
+
+        // await rpc endpoint to finish to ensure we have the rpc endpoints before we
+        // update the chain resources
+        retrieveRpcEndpointsDeferred.await()
+        val chainsResponse = chainsRequestDeferred.await()
+
+        if (chainsResponse != null) {
+            update(stateMachine.routerChains(chainsResponse), oldState)
+        }
+    }
+
+    private suspend fun retrieveChainRpcEndpoints() = suspendCancellableCoroutine<Unit> { continuation ->
+        val url = "${helper.deploymentUri}/configs/rpc.json"
+        helper.get(url) { _, response, _, _ ->
+            if (response != null) {
+                try {
+                    Json.decodeFromString<Map<String, RpcInfo>>(response).let {
+                        RpcConfigs.chainIdToRpcMap = it
+                    }
+                } catch (e: IllegalArgumentException) {
+                    Logger.e { "retrieveChainRpcEndpoints IllegalArgumentException error: $e" }
+                    throw e  // Rethrow if necessary
```

Comment:
wait we explicitly do not want to rethrow.

Reply:
894b4f2515986d3c5be0b3cda66234a6917b4328